### ISSUE TITLE
[BUG]: Normalization or Transformation works incorrectly when data contains categorical variable

### ISSUE THREAD

**Issue by Azlich** [pycaret/pycaret#3076]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

trying to follow this [link](https://www.learndatasci.com/tutorials/introduction-pycaret-machine-learning/)

In PyCaret 3.0.0rc4, the setup steps when I specify `normalize=True` or `tranformation=True`. It transforms all variables even though I set the column name in `categorical_features` but it works well on PyCaret 2.3.10 with the same source code

### Reproducible Example

```python
from pycaret.datasets import get_data
from pycaret.regression import *

data = get_data('insurance')

setup(
    data=data, 
    target='charges', 
    normalize=True, 
    numeric_features=['age', 'children', 'bmi'], 
    categorical_features=['sex', 'smoker', 'region'],
    session_id=101
)

get_config('X_train')
```

### Expected Behavior

![pycaret2 3 10](https://user-images.githubusercontent.com/16594006/200835818-261697d8-bf14-4908-8c03-7dd073c7fd0a.png)

### Actual Results

```python-traceback
![pycaret3 0 0rc4](https://user-images.githubusercontent.com/16594006/200835460-94a4d33d-8009-4abf-8a9e-0e6073af762a.png)
```

### Installed Versions

<details>
System:
    python: 3.7.15 (default, Oct 12 2022, 19:14:55)  [GCC 7.5.0]
executable: /usr/bin/python3
   machine: Linux-5.10.133+-x86_64-with-Ubuntu-18.04-bionic

PyCaret required dependencies:
                 pip: 21.1.3
          setuptools: 57.4.0
             pycaret: 3.0.0rc4
             IPython: 7.9.0
          ipywidgets: 7.7.1
                tqdm: 4.64.1
               numpy: 1.21.6
              pandas: 1.3.5
              jinja2: 3.0.0
               scipy: 1.7.3
              joblib: 1.2.0
             sklearn: 1.0.2
                pyod: 1.0.6
            imblearn: 0.8.1
   category_encoders: 2.5.1.post0
            lightgbm: 3.3.3
               numba: 0.55.2
            requests: 2.28.1
          matplotlib: 3.5.3
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.5.0
             kaleido: 0.2.1
         statsmodels: 0.12.2
              sktime: 0.13.4
               tbats: 1.1.1
            pmdarima: 1.8.5
              psutil: 5.9.4

PyCaret optional dependencies:
                shap: Not installed
           interpret: Not installed
                umap: Not installed
    pandas_profiling: 1.4.1
  explainerdashboard: Not installed
             autoviz: Not installed
           fairlearn: Not installed
             xgboost: 0.90
            catboost: Not installed
              kmodes: Not installed
             mlxtend: 0.14.0
       statsforecast: Not installed
        tune_sklearn: Not installed
                 ray: Not installed
            hyperopt: 0.1.2
              optuna: Not installed
               skopt: Not installed
              mlflow: Not installed
              gradio: Not installed
             fastapi: Not installed
             uvicorn: Not installed
              m2cgen: Not installed
           evidently: Not installed
                nltk: 3.7
            pyLDAvis: Not installed
              gensim: 3.6.0
               spacy: 3.4.2
           wordcloud: 1.8.2.2
            textblob: 0.15.3
               fugue: Not installed
           streamlit: Not installed
             prophet: 1.1.1
</details>

**Comment by moezali1** [comment]
@tvdboom Not sure if there is any background to this, but have we intentionally chosen to apply normalization and transformation on categorical features as well?

**Comment by tvdboom** [comment]
Normalization is applied through sklearn's [StandardScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html), which transforms all numerical features. Since the features sex, smoker and region have been encoded before, they are no longer categorical and are therefore transformed. This is intended behavior and definitely not a bug since it doesn't influence the pipeline's performance. Is there any reason why these features must remain not-normalized @Azlich ?

**Comment by Azlich** [comment]
@tvdboom  At first, I thought it should get the same result as PyCaret 2.3.10. After I did some research I found it has both scaling and not IMO I think it should allow users to choose.

reference from [here](https://stats.stackexchange.com/questions/59392/should-you-ever-standardise-binary-variables)
